### PR TITLE
Fixed the multi-gpu context

### PR DESCRIPTION
**Pull request by karan6181**
- Fixed the multi-GPU context where the test now utilizes a user-defined number of GPUs

For more information, see this [issue](https://github.com/awslabs/keras-apache-mxnet/issues/67)

@sandeep-krishnamurthy @roywei

### CODE REVIEW

**Comment by roywei** [comment]
@kalyc could you also take a look? thanks!

File: keras/models.py [hunk 1]
```diff
@@ -577,8 +577,15 @@ def build(self, input_shape=None):
             raise TypeError('Sequential model cannot be built: model is empty.'
                             ' Add some layers first.')
         # actually create the model
-        self.model = Model(self.inputs, self.outputs[0],
-                           name=self.name + '_model')
+        if K.backend() == 'mxnet' and hasattr(self, '_context') and \
+                self._context is not None:
```
Comment (kalyc) [inline keras/models.py]: can use short circuit here - move self._context is not None condition first and follow with the rest

Comment (karan6181) [reply keras/models.py]: It fails if `_context` parameter is not part of an object(self or sequential() here), that's why I am checking first whether `_context` parameter is available as a class variable or not. Based on the different backend, `self._context` parameter is not added in the constructor.

Comment (kalyc) [reply keras/models.py]: Makes sense. Thanks!

**Comment by karan6181** [comment]
When you call `keras.utils.multi_gpu_model(model, gpus=gpus)`, the `model` refer to as `keras.models.Sequential object` and it sets the appropriate `context`. Here in this case it's `[gpu(0), gpu(1), gpu(2), gpu(3)]` if the number of GPUs is 4.

During compile phase, Sequential build() method(`def build() in models.py`) creates a MXNet model object(`keras.backend.mxnet_backend.get_model`) and this MXNet model set context as default context(`[gpu(0)]`) and calls `self.model.compile()`.

So, there is a separate context for each model heirarchy and both are not in sync. First context is set during `multi_gpu_model` for `keras.models.Sequential object` (Which set correct context) and second context is set during `build()` for  `keras.backend.mxnet_backend.get_model`(which set incorrect context).



Note:

```python
self: <keras.models.Sequential object at 0x7f3002ad9630>

self.model: <keras.backend.mxnet_backend.get_model.<locals>.Model object at 0x7f2ffb4ec9b0> 
```

**Pull request merged**